File: Assignment GTR Ltd/Frontend Angular/app/controllers/productdetail.js
var app = angular.module('my-app', []);
app.controller('detailCtrl',($scope,$http,$window)=>{
  $scope.productdetail = ()=>{
    var data = 
    {
        PurchaseCustomer:$scope.PurchaseCustomer, 
        PurchaseDate:$scope.PurchaseDate,
        SalesEmployee:$scope.SalesEmployee, 
        Description:$scope.Description,
        Stock:$scope.Stock, 
        Status:$scope.Status,
        ProductMasterId:$scope.ProductMasterId, 
    };
    $http.post("https://localhost:7180/api/productdetail/add",data).then(function(res){
            alert("Product created successfully");
            $window.location.href = 'productdetailList.html';
            
        }).catch(function(err){
            //debugger;
            alert("Created failed incorrect information");
            //console.error('Error saving data:', err);
        });
   
  };
});

File: Assignment GTR Ltd/Asp.net Core Web API Assignment Backend/Asp.net Core Web API Assignment Backend/Models/ProductDetail.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Asp.net_Core_Web_API_Assignment_Backend.Models
{
    public class ProductDetail
    {
        public int Id { get; set; }
        public string PurchaseCustomer { get; set; }
        public string PurchaseDate { get; set; }
        public string SalesEmployee { get; set; }
        public string Description { get; set; }
        public int Stock { get; set; }
        public string Status { get; set; }
        [ForeignKey("ProductMaster")]
        public int ProductMasterId { get; set; } 
        //public virtual ProductMaster ProductMaster { get; set; }
    }
}
